packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +570,655 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+/// then the divider above and below that pressed button are not drawn - instead
+/// they are filled with the standard white dialog background color. The one
+/// exception is the very 1st divider which is always rendered. This policy comes
+/// from observation of native iOS dialogs.
+@visibleForTesting
+class RenderCupertinoDialogActions extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, MultiChildLayoutParentData>,
+        RenderBoxContainerDefaultsMixin<RenderBox, MultiChildLayoutParentData> {
+  /// Creates an iOS dialog actions section layout policy.
+  ///
+  /// [pressedButtons] must not be null.
+  RenderCupertinoDialogActions({
+    List<RenderBox> children,
+    @required Set<int> pressedButtons,
+    double dividerThickness = 0.0,
+  }) : assert(pressedButtons != null),
+        _dividerThickness = dividerThickness,
+        _pressedButtons = pressedButtons {
+    addAll(children);
+  }
+
+  Set<int> _pressedButtons;
+
+  /// The buttons which are currently pressed by the user.
+  Set<int> get pressedButtons => _pressedButtons;
+
+  set pressedButtons(Set<int> newValue) {
+    if (_pressedButtons.containsAll(newValue)
+        && newValue.containsAll(_pressedButtons)) {
+      return;
+    }
+
+    _pressedButtons = newValue;
+    markNeedsPaint();
+  }
+
+  double _dividerThickness;
+
+  /// The thickness of the divider between buttons.
+  double get dividerThickness => _dividerThickness;
+
+  set dividerThickness(double newValue) {
+    if (newValue == _dividerThickness) {
+      return;
+    }
+
+    _dividerThickness = newValue;
+    markNeedsLayout();
+  }
+
+  final Paint _buttonBkPaint = new Paint()
+    ..color = _kDialogColor
+    ..style = PaintingStyle.fill;
+
+  final Paint _pressedButtonBkPaint = new Paint()
+    ..color = _kDialogPressedColor
+    ..style = PaintingStyle.fill;
+
+  final Paint _dividerPaint = new Paint()
+    ..color = _kButtonDividerColor
+    ..style = PaintingStyle.fill;
 
   @override
-  void paint(Canvas canvas, Size size) {
-    final Paint paint = new Paint()..color = _kButtonDividerColor;
-    canvas.drawLine(const Offset(0.0, 0.0), new Offset(size.width, 0.0), paint);
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! MultiChildLayoutParentData)
+      child.parentData = new MultiChildLayoutParentData();
   }
 
   @override
-  bool shouldRepaint(_CupertinoVerticalDividerPainter other) => false;
-}
+  double computeMinIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    if (childCount == 0) {
+      return 0.0;
+    } else if (childCount == 1) {
+      // If only 1 child
+      return _computeMinIntrinsicHeightSideBySide(width);
+    } else {
+      final List<RenderBox> children = getChildrenAsList();
+
```

Comment:
why the blank line?

Reply:
Because it separates a declaration from a series of conditionals...I find that much easier to read and I tend to write code that way in general.  Would you like me to get rid of it?

Reply:
It seems weird to me because I read the code in blocks separated by blank lines, and this is implying that the code on either side of the "else" is more closely related than the two parts of the code within the "else". But it's up to you.